integrationTests/chainSimulator/mempool/testutils_test.go

```diff
@@ -523,10 +553,45 @@ func testSecondSelectionWithManyTxsInPool(t *testing.T, sw *core.StopWatch, numT
 	)
 	require.Nil(t, err)
 
+	f := createPprofFiles(t)
+	if f != nil {
+		defer func() {
+			// stop profiling
+			pprof.StopCPUProfile()
+			err = f.Close()
+			require.NoError(t, err)
+		}()
+	}
+
 	// measure the time for the second selection (now we use the breadcrumbs to create the virtual records)
 	sw.Start(t.Name())
 	selectedTransactions, _ = txpool.SelectTransactions(selectionSession, options, blockchainInfo)
 	sw.Stop(t.Name())
 
 	require.Equal(t, numTxsToBeSelected, len(selectedTransactions))
 }
+
+func createPprofFiles(t *testing.T) *os.File {
+	shouldCreatePprofFiles := checkEnvVariables()
+	if !shouldCreatePprofFiles {
+		return nil
+	}
+
+	// start profiling
+	testName := strings.ReplaceAll(t.Name(), "/", "_")
+	fileName := fmt.Sprintf("%s.pprof", testName)
+	f, err := os.Create(fileName)
+	require.Nil(t, err)
+
+	err = pprof.StartCPUProfile(f)
+	require.Nil(t, err)
+
+	return f
+}
+
+func checkEnvVariables() bool {
```

Comment:
Function has a generic name, does something specific. Maybe rename it to `shouldCreatePprofFiles()`?